include/arch/arc/v2/arcv2_irq_unit.h

```diff
@@ -189,19 +215,38 @@ bool z_arc_v2_irq_unit_is_in_isr(void)
  *
  * @return N/A
  */
+static ALWAYS_INLINE
+void z_arc_v2_irq_unit_trigger_set(int irq, unsigned int trigger)
+{
+	unsigned int key = arch_irq_lock();
 
-void z_arc_v2_irq_unit_trigger_set(int irq, unsigned int trigger);
```

Comment:
Please check intc_arcv2_irq_unit.c as well - that has an implementation of this and other functions as well that do not lock interrupts. (looks like a mistake introduced when the timer API was updated: duplicate inline implementation in the header).